src/plugins/__mocks__/mocks.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { dataPluginMock } from '../data/public/mocks';
+
+export const pluginsMock = {
```

Comment:
I was thinking it could simplify testing if there was one object with all plugin mocks, similar to how there is `coreMock` with mocks of all Core services.

> ...instead people would import mocks for core and whichever services their plugin depends on.

Yes, you still need to explicitly import it in the New Platform. The idea is&mdash;if your plugin depends on 5 other plugins it is more convenient to import an object of all plugin mocks from one place then have 5 different `import` statements and construct that object yourself. If you want to be really precise in your tests, you can destructure that object and pick only the needed plugin mocks.

Reply:
I think what @lukeelmers is suggesting is not to change any functionality but just to move the contents of this file into `src/legacy/ui/public/new_platform/__mocks__/helpers.ts`. I think that's a good idea :+1: Like you said, it makes sense to get all the plugin mocks when you're consuming NP from legacy `ui/new_platform` but if all the mocking logic lives in the same file, then we can just delete that file once the migration is complete.